ckan/logic/action/update.py

```diff
@@ -62,6 +62,8 @@ def resource_update(context, data_dict):
     model = context['model']
     user = context['user']
     id = _get_or_bust(data_dict, "id")
+    old_resource = _get_action('resource_show')(context, {'id': id})
+
```

Comment:
I think that you should allow CKAN to perform all its default checks first and obtain old information no sooner than required